### ISSUE TITLE
🐛 Biome encountered an unexpected error

### ISSUE THREAD

**Issue by atheck** [biomejs/biome#564]
### Environment information

```block
CLI:
  Version:                      1.3.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.18.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.1.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

Source Location: crates/biome_js_analyze/src/utils.rs:110:13
Thread Name: biome::worker_7
Message: internal error: entered unreachable code: The node can't have the same range of the operator.

### Expected result

No error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Thank you, @atheck for the report. Are you able to narrow down the cause of this issue to a single snippet?

**Comment by atheck** [comment]
I'm using `extends` in the biome.json with a custom set of linting rules. It works when I copy those linting rules to the biome.json itself and remove the `extends`.

Not working:

```json
{
  "extends": ["./lintingRules.json"]
}
```

Works:

```json
{
  "linter": {
    "enabled": true,
    "rules": {
      // same rules as in lintingRules.json
    }
  }
}
```

**Comment by ematipico** [comment]
This should be fixed, it will be available in next release
